### PR TITLE
Implement get api which returns provider handles, given git alias

### PR DESCRIPTION
**Pull request by tapishr**
- [x] Build running, manually tested
- [x] Corresponding DPU PR - [https://github.com/Alokit-Innovations/vibi-dpu/pull/119](https://github.com/Alokit-Innovations/vibi-dpu/pull/119)

## Summary by CodeRabbit

- **New Features**
	- Added `aliasesPostHandler` to handle POST requests.
	- Added `aliasesGetHandler` to handle GET requests with query parameter validation.
	- Updated `aliasesHandler` to route requests based on HTTP method.

### CODE REVIEW

File: pages/api/dpu/aliases.ts [hunk 1]
```diff
@@ -27,4 +40,30 @@ const aliasesHandler = async (req: NextApiRequest, res: NextApiResponse) => {
...
+    }
+
+    // Retrieve aliases from the database
+    try {
+        const aliases = await getUserAliasesFromDb(
+            repo_name as string, repo_owner as string, repo_provider as string
+        );
+        console.info("[aliasesGetHandler] Aliases retrieved successfully");
+        res.status(200).json({ aliases });
+    } catch (error) {
```
Comment (MuskanPaliwal) [inline pages/api/dpu/aliases.ts:63]: remove try-catch and use .catch on await statement

File: utils/db/aliases.ts [hunk 2]
```diff
@@ -24,3 +24,27 @@ export const saveUserAliasesToDb = async (repo_name: string, repo_owner: string,
...
+            AND r.repo_provider = $3;
+    `;
+    try {
+        const { rows } = await conn.query(query, [repo_name, repo_owner, repo_provider]);
+        const aliases = rows.map(row => ({
+            git_alias: row.git_alias,
+            [handleColumn]: row[handleColumn]
+        }));
+        return aliases;
+    } catch (error) {
```
Comment (MuskanPaliwal) [inline utils/db/aliases.ts:46]: same as above

**Review by MuskanPaliwal — CHANGES_REQUESTED** [review]
good to go from my end

File: utils/db/aliases.ts (hunk 2 above)
Comment (tapishr) [reply utils/db/aliases.ts]: done

File: pages/api/dpu/aliases.ts (hunk 1 above)
Comment (tapishr) [reply pages/api/dpu/aliases.ts]: done

File: pages/api/dpu/aliases.ts [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+      } else if (req.method === 'POST') {
+        // Handle POST request
+        await aliasesPostHandler(req, res);
+      } else {
+        // Handle other request methods
+        res.status(405).end(); // Method Not Allowed
+      }
+}
+
+const aliasesPostHandler = async (req: NextApiRequest, res: NextApiResponse) => {
```
[the remark this answers is not in the document]
Comment (MuskanPaliwal) [reply pages/api/dpu/aliases.ts:17]: rabbit is correct here @tapishr

**Review by MuskanPaliwal — APPROVED** [review]

**Review by MuskanPaliwal — APPROVED** [review]

**Pull request merged**